awx/api/views/__init__.py

```diff
@@ -451,28 +456,12 @@ def post(self, request, *args, **kwargs):
                 if extra_field in runner_data:
                     data[extra_field] = runner_data[extra_field]
```

Comment:
You're not doing anything with `data` anymore.

You're also calling `execution_node_health_check(obj.hostname)`, not `execution_node_health_check.delay(obj.hostname)`, so it runs instantaneously. The point of `data` was to return the data _from_ the health check.

The response message _suggests_ that it is ran asynchronously, but you need to do a `.delay` call for that to be true. There should be an upper limit to how long it can take (20 or 40 seconds), which is why this was done synchronously to begin with, for background.

Reply:
@AlanCoding I had done a bad rebase.  I have since fixed that and updated this PR.  Please re-review